src/command/build.rs

```diff
@@ -378,6 +387,9 @@ impl Build {
     }
 
     fn step_run_wasm_opt(&mut self) -> Result<(), Error> {
+        if self.reference_types {
+            return Ok(());
```

Comment:
This should generate at least a debug message to notify the user that wasm-opt had been disabled.